lib/Doctrine/Common/Persistence/AbstractManagerRegistry.php

```diff
@@ -210,7 +210,17 @@ public function getManagers()
      */
     public function getRepository($persistentObjectName, $persistentManagerName = null)
     {
-        return $this->getManager($persistentManagerName)->getRepository($persistentObjectName);
+        if ($persistentManagerName !== null) {
+            $manager = $this->getManager($persistentManagerName);
+        } else {
```

Comment:
Could we avoid `else` branch and nested  `if` inside? Possibly using private method and early-exit approach.